mesonbuild/dependencies/dev.py

```diff
@@ -171,9 +170,33 @@ def __init__(self, environment, kwargs):
         else:
             self._set_old_link_args()
         self.link_args = strip_system_libdirs(environment, self.link_args)
+        self.link_args = self.__strip_bogus_link_args(self.link_args)
+
+    @staticmethod
+    def __strip_bogus_link_args(args):
+        """Remove bogus linker args that llvm-config returns.
+
+        FreeBSD's llvm-config returns -l/usr/lib/libexecinfo.so
+        """
+        new_args = []
+        for arg in args:
+            if arg.endswith('.so'):
```

Comment:
Maybe this should be `arg.startswith('-l') and arg.endswith('.so')`? Because it would be perfectly valid to return the full path to a .so file as the link argument, right? llvm-config may not do so now, but still...

Reply:
That's actually what it does and the linker doesn't like that.

Reply:
specifically `-l/usr/lib/libexecinfo.so`

Reply:
Yes, that's an invalid argument. To link to a `.so`, you give `/usr/lib/libexecinfo.so` not `-l/usr/lib/libexecinfo.so`.